students/daniel_carrasco/session04/trigrams.py

```diff
@@ -0,0 +1,72 @@
+import sys
+import re
+import random
+
+def read_in_data(filename):
+    f = open(filename)
+    lines=[]
+    while True:
+        line = f.readline()
+        if not line:
+            break
+        lines.append(line.strip('\n'))
+    return lines
+
+def make_words(in_data):
+    new_data=[]
+    for line in in_data:
+        line = re.sub(r'[^\w\s]',',',line)
+        line = re.sub(r',,',' ',line)
+        line = re.sub(r',',' ',line)
+        line = re.sub(r'  ',' ',line)
```

Comment:
I think it might be easier to use `.replace()` on the line to get rid of punctuation and spaces rather than regex